pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -935,14 +939,9 @@ protected boolean commitSegment(String controllerVipUrl, boolean isSplitCommit)
     return segmentCommitter.commit(_segmentBuildDescriptor);
   }
 
-  protected boolean buildSegmentAndReplace() {
+  protected void buildSegmentAndReplace() throws Exception {
```

Comment:
(code format) Please setup the code format following: https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#setup-ide